src/App/Repository/UserRepository.php

```diff
@@ -188,9 +190,7 @@ public function register(Entity $entity)
 	// ResetPasswordRepository
 	public function getResetToken(Entity $entity)
     {
-		// Todo: replace with something more robust.
-		// This is predictable if we don't have the openssl mod
-		$token = Security::token(true);
+		$token = Hash::make(Str::random(40));
```

Comment:
I'm not really sure if hashing is necessary here, since I'm not verifying the hash. We could probably just use a 40 character random string.

The laravel built in reset tokens use a hash, but they also verify it. We could do that too, ie:
- Generate a string.
- Hash it and save that to the db
- Return the original token string to the caller (ie. usecase that then sends an email)
- Update our verification to verify the hash, not simply a token match

Reply:
So I think like with reset tokens it's best to gen a really really really random string, then save the hash in the db. This is probably best so that the token can't just be read out of the DB. We need to make sure that random() is legitimately random. I think we need to make sure openssl_random_pseudo_bytes is installed otherwise rand just uses quickRandom which is a bit meh.

So that's a long way of saying, yeah the above list of steps sounds good.